crates/bins/src/bin/datadog-static-analyzer-git-hook.rs

```diff
@@ -448,31 +462,47 @@ fn main() -> Result<()> {
         all_rule_results.extend(rule_results);
     }
 
-    for rule_result in all_rule_results {
-        let path = PathBuf::from(rule_result.filename);
-
-        for violation in rule_result.violations {
-            if let Some(lines) = modifications.get(&path) {
-                if lines.contains(&violation.start.line) {
-                    println!(
-                        "{}",
-                        format_error(
-                            path.display().to_string().as_str(),
-                            violation.start.line,
-                            &rule_result.rule_name,
-                            IssueType::StaticAnalysis,
-                        )
-                    );
-                    fail_for_static_analysis = true;
-                    continue;
-                }
-            }
+    // Filter the results and keep only the results relevant for the diff
+    let results_for_diff: Vec<RuleResult> = all_rule_results
+        .into_iter()
+        .map(|rr| {
+            let mut new_result = rr.clone();
+            let filtered_violations = rr
+                .violations
+                .into_iter()
+                .filter(|v| {
+                    let path = PathBuf::from(&rr.filename);
+                    if let Some(lines) = modifications.get(&path) {
+                        lines.contains(&v.start.line)
+                    } else {
+                        false
+                    }
+                })
+                .collect();
+            new_result.violations = filtered_violations;
+            new_result
+        })
+        .collect();
```

Comment:
I don't think it makes sense to consume an iterator and then clone each element inside `map`, we should mutate the vec in place instead. You can do this with `iter_mut` + `for_each` or just do `for rr in &mut all_rule_results`, and then in the body you modify the `violations` field with `retain` to retain only the elements that satisfy the clojure. Or you could also perform the filtering logic *before* you extend `all_rule_results`. My suggestion uses `iter_mut` + `for_each` with `retain`, but I think it'd be best to do this before you extend `all_rule_results` above.

```suggestion
    all_rule_results.iter_mut().for_each(|rr| {
        let path = PathBuf::from(&rr.filename);
        rr.violations.retain(|v| {
            if let Some(lines) = modifications.get(&path) {
                lines.contains(&v.start.line)
            } else {
                false
            }
        });
    });
```